### DISCUSSION TITLE
TUI issues - locked out and stuck in loop

### DISCUSSION THREAD

**Discussion by corriganjeff** [JanssenProject/jans#6553] [Q&A]
Is there an auth cache for the TUI? I seem to be stuck in a loop. I was able to login yesterday and prompted for device code flow. Today though I get the error screen that the authorization grant is invalid/expired/revoked. On the next screen with the hostname/clientid/secret I hit save and then get thrown back to the previous error screen. 

I can see in the auth log that the tui client successfully authenticated. Is there a cache or something I can clear to get past this issue and force a restart of the device code flow?

**Comment by corriganjeff** [comment]
@ossdhaval  Any thoughts? I think this is a bug or an issue with documentation. I made a systemic changes to force ATs as jwt and disabled implicit flow and did some random excludes to keep some acrs out of the discovery doc before having this where the tui reports that it failed to get an access token. I didn't make an explicit record of every change made as I was not expecting to have issues with basic config changes.

**Comment by ossdhaval** [comment]
Hi @corriganjeff `Logout and exit` option should clear the state. Can you see if this takes care of the scenario you faced?

![image](https://github.com/JanssenProject/jans/assets/343411/fcff9490-817b-48a0-a563-1d92a57a2bd6)

**Comment by devrimyatar** [comment]
*Logout and Exit Jans TUI* should work. If not, please check client and secret. Maybe one of them changed somehow.

**Comment by nynymike** [comment]
If all else fails, you can `rm` the file `.config/jans-cli.ini`

**Reply by corriganjeff** [reply]
I blew the instance away and restarted. I think it happened after I disabled implicit grant for the entire idp. I modified the tui's python file with some echos so I could troubleshoot the failure. On the backend I could see the authentication was successful but the tui was failing to get back access tokens. If I come across the issue again I'll try the logout method you pointed out. Thanks!
